doc/string/slices.rdoc

```diff
@@ -0,0 +1,144 @@
+== \String Slices
+
+A _slice_ of a string is a substring that is selected by certain criteria.
+
+These instance methods make use of slicing:
+
+- String#[] (and its alias String#slice) return a slice copied from +self+.
+- String#[]= returns a copy of +self+ with a slice replaced.
+- String#slice! returns +self+ with a slice replaced.
```

Comment:
I think `String#slice!` removes a slice.

For example:

```ruby
irb(main):010:0> str = "hello"
=> "hello"
irb(main):011:0> str.slice!(2..3)
=> "ll"
irb(main):012:0> str
=> "heo"
```

Reply:
Fixed.